docs/basics/networkbehaviour.md

```diff
@@ -108,3 +108,130 @@ If you override the virtual 'OnDestroy' method it is important to alway invoke t
+In the example below, it provides one simple use-case scenario where you can use `NetworkBehaviour.OnSynchronize` to synchronize late-joining clients with state set by `ClientRpc` events:
+```csharp
+using UnityEngine;
+using Unity.Netcode;
+
+/// <summary>
+/// Simple RPC driven state that demonstrates one
+/// form of NetworkBehaviour.OnSynchronize usage
+/// </summary>
+public class SimpleRpcState : NetworkBehaviour
+{
+    private bool m_ToggleState;
+
+    /// <summary>
+    /// Late joining clients will be synchronized 
+    /// to the most current m_ToggleState
+    /// </summary>
+    protected override void OnSynchronize<T>(ref BufferSerializer<T> serializer)
+    {
+        serializer.SerializeValue(ref m_ToggleState);
+        base.OnSynchronize(ref serializer);
+    }
+
+    public void ToggleState(bool stateIsSet)
+    {
+        m_ToggleState = stateIsSet;
+    }
+
+    /// <summary>
+    /// Synchronizes connected clients with the
+    /// server-side m_ToggleState
+    /// </summary>
+    /// <param name="stateIsSet"></param>
+    [ClientRpc]
+    private void ToggleStateClientRpc(bool stateIsSet)
+    {
+        m_ToggleState = stateIsSet;
+    }
+}
+```
+:::caution
+Since `NetworkBehaviour.OnSynchronize` is primarily used for server to client synchronization, RPC state synchronization only works when using ClientRpcs since `NetworkBehaviour.OnSynchronize` is only invoked on the server side during the write portion of serialization and only invoked on the client side during the read portion of serialization. When running a host `NetworkBehaviour.OnSynchronize` is still only invoked once (server-side) during the write portion of serialization.
+:::
+
+### Debugging OnSynchronize Serialization
+If your serialization code has a bug and throws an exception, then `NetworkBehaviour.OnSynchronize` has additional safety checking to handle a graceful recovery without completely breaking the rest of the synchronization serialization pipeline.  
+
+#### When Writing:
+If user-code throws an exception during `NetworkBehaviour.OnSynchronize`, it catches the exception and if:
+- **LogLevel = Normal**: A warning message that includes the name of the `NetworkBehaviour` that threw an exception while writing will be logged and that portion of the serialization for the given `NetworkBehaviour` is skipped.
+- **LogLevel = Developer**: It provides the same warning message as well as it logs an error with the exception message and stack trace.
+
+After generating the log message(s), it rewinds the serialization stream to the point just before it invoked `NetworkBehaviour.OnSynchronize` and will continue serializing. Any data written before the exception occurred will be overwritten or dropped depending upon whether there are more `NetworkBehaviour` components to be serialized.
+
```

Comment:
`NetworkBehaviour.OnSynchronize` has additional safety checking to handle a graceful recovery without completely breaking the rest of the synchronization serialization pipeline if your code throws an error.

#### When Writing:
If user code throws an exception during `NetworkBehaviour.OnSynchronize`, it catches the exception. The way it handles the exception depends on the `LogLevel`.
- **LogLevel = Normal**: It logs a warning message that includes the name of the `NetworkBehaviour` that threw an exception while writing and skips the portion of the serialization for the given `NetworkBehaviour`.
- **LogLevel = Developer**: It provides the same warning message and logs an error with the exception message and stack trace.

After generating the log message(s), it rewinds the serialization stream to the point just before invoking `NetworkBehaviour.OnSynchronize` and continues serializing. It overwrites or drops any data written before the exception occurred (depending on whether it has more `NetworkBehaviour` components to serialize).